tests/disaster-recovery/regional-dr/test_failover.py

```diff
@@ -0,0 +1,93 @@
+import logging
+import pytest
+
+from time import sleep
+
+from ocs_ci.framework import config
+from ocs_ci.framework.testlib import rdr_test
+from ocs_ci.helpers import dr_helpers
+from ocs_ci.ocs.exceptions import CommandFailed
+from ocs_ci.ocs.node import wait_for_nodes_status, get_node_objs
+from ocs_ci.ocs.ocp import wait_for_cluster_connectivity
+
+logger = logging.getLogger(__name__)
+
+
+@rdr_test
+class TestFailover:
+    @pytest.fixture()
+    def setup(self, request, nodes, rdr_workload):
+        dr_helpers.set_primary_cluster_context(rdr_workload.workload_namespace)
+        primary_cluster_index = config.cur_index
+        all_nodes = get_node_objs()
+
+        def finalizer():
+            config.switch_ctx(primary_cluster_index)
+            try:
+                wait_for_cluster_connectivity(tries=3)
+            except CommandFailed:
+                nodes.start_nodes(all_nodes)
+                wait_for_nodes_status([node.name for node in all_nodes])
+
+        request.addfinalizer(finalizer)
+
+        return all_nodes, rdr_workload
+
+    @pytest.mark.parametrize(
+        argnames=["primary_cluster_down"],
+        argvalues=[
+            pytest.param(False),
+            pytest.param(True),
+        ],
+    )
+    def test_failover(self, nodes, setup, primary_cluster_down):
+        """
+        Test Failover action
+
+        """
+        all_nodes, rdr_workload = setup
+
+        scheduling_interval = dr_helpers.get_scheduling_interval(
+            rdr_workload.workload_namespace
+        )
+        logger.info(f"Waiting for {2 * scheduling_interval} minutes for IOs to run")
+        sleep(2 * scheduling_interval * 60)
+
+        if primary_cluster_down:
+            dr_helpers.set_primary_cluster_context(rdr_workload.workload_namespace)
+            nodes.stop_nodes(all_nodes)
+
+        secondary_cluster_name = dr_helpers.get_secondary_cluster_name(
+            rdr_workload.workload_namespace
+        )
+        dr_helpers.failover(secondary_cluster_name, rdr_workload.workload_namespace)
+
+        # Verify workload deployment on failedover cluster
+        dr_helpers.set_primary_cluster_context(rdr_workload.workload_namespace)
```

Comment:
This is supposed to be secondary cluster context right ? Am I missing something ?

Reply:
Once workload is moved to another cluster, the secondary cluster will become primary cluster w.r.t to that workload.
So for ex. When we do failover from C1 to C2, the secondary cluster C2 will now become primary cluster for the particular workload.